test/src/base/MooseTestApp.C

```diff
@@ -141,6 +144,12 @@ MooseTestApp::registerApps()
   registerApp(MooseTestApp);
 }
 
+std::string
+MooseTestApp::header() const
+{
+  return getParam<std::string>("append_header");
```

Comment:
```suggestion
  if (isParamSetByUser("append_header"))
    return getParam<std::string>("append_header");
```

Reply:
Still need to return something. I thought the default was an appropriate way for this?

Reply:
Or just `return "";` on a new line.